easybuild/easyconfigs/s/SciPy-bundle/scipy-1.11.1_meson-build-mkl-name.patch

```diff
@@ -0,0 +1,20 @@
+Meson tries to find MKL by using 'mkl' as a package name to pkg-config, that does not work.  The real error is probably that -Dblas=mkl -Dlapack=mkl is passed to the meson build instead of -Dblas=mkl-dynamic-lp64-seq etc.  The simplest fix is to hack the meson.build script.
+
+Author: Jakob Schiotz (Techn. Univ. Denmark) <schiotz@fysik.dtu.dk>
+
+--- scipy/meson.build.old	2023-10-24 08:39:29.966344750 +0200
++++ scipy/meson.build	2023-10-24 09:08:24.832773698 +0200
+@@ -153,6 +153,13 @@
+ # https://github.com/mesonbuild/meson/issues/2835
+ blas_name = get_option('blas')
+ lapack_name = get_option('lapack')
++# Hack for MKL in easybuild
++if blas_name == 'mkl'
++  blas_name = 'mkl-dynamic-lp64-seq'
++endif
++if lapack_name == 'mkl'
++  lapack_name = 'mkl-dynamic-lp64-seq'
++endif
```

Comment:
I think we should just fix the scipy easyblock instead. The most annoying part would just be to check if this behavior is backwards compatible with older scipy's or if we need to make a check with LooseVersion and only use the full "mkl-dynamic-lp64-seq" name going forward.

I also wonder how this plays together with RPATH users .. sigh. Maybe more involved logic that uses the environment variables that the toolchain sets up should be attempted instead of these prepackages PC files.

Reply:
scipy only builds with Meson from ``SciPy-bundle-2023.02`` and later, and there are no Intel versions of these, so I do not think there will be any problem with backwards compatibility.

I have no clue about RPATH, but usually pkg-config sets that correctly.